crates/viewer/re_ui/src/notifications.rs

```diff
@@ -0,0 +1,393 @@
+            if notifications.is_empty() {
+                ui.label(
+                    egui::RichText::new("Nothing here!")
+                        .weak()
+                        .color(hex_color!("#636b6f")),
+                );
+
+                return;
+            }
+
+            for (i, notification) in notifications.iter().enumerate().rev() {
+                show_notification(ui, notification, DisplayMode::Panel, || {
+                    to_dismiss = Some(i);
+                });
+            }
+        };
+
+        let mut clear = false;
+
+        egui::Area::new(self.id)
+            .anchor(egui::Align2::RIGHT_TOP, egui::vec2(-8.0, 32.0))
+            .order(egui::Order::Foreground)
+            .interactable(true)
+            .movable(false)
+            .show(egui_ctx, |ui| {
+                egui::Frame::window(ui.style())
+                    .fill(hex_color!("#141819"))
+                    .rounding(8.0)
+                    .inner_margin(8.0)
+                    .show(ui, |ui| {
+                        ui.set_width(panel_width);
+                        ui.horizontal_top(|ui| {
+                            ui.label("Notifications");
+                            ui.with_layout(egui::Layout::top_down(egui::Align::Max), |ui| {
+                                if ui.small_icon_button(&icons::CLOSE).clicked() {
+                                    *is_panel_visible = false;
+                                }
+                            });
+                        });
+                        egui::ScrollArea::vertical()
+                            .min_scrolled_height(panel_max_height)
+                            .show(ui, notification_list);
+
+                        if !notifications.is_empty() {
+                            ui.horizontal_top(|ui| {
+                                if ui.button("Clear all").clicked() {
+                                    clear = true;
+                                };
+                            });
+                        }
+                    });
+            });
+
+        if clear {
+            notifications.clear();
+        } else if let Some(to_dismiss) = to_dismiss {
+            notifications.remove(to_dismiss);
+        }
+    }
+}
+
+fn base_ttl() -> Duration {
+    Duration::from_secs_f64(4.0)
+}
+
+struct Toasts {
+    id: egui::Id,
+}
+
+impl Default for Toasts {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Toasts {
+    fn new() -> Self {
+        Self {
+            id: egui::Id::new("__toasts"),
+        }
+    }
+
+    /// Shows and updates all toasts
+    fn show(&self, egui_ctx: &egui::Context, notifications: &mut [Notification]) {
+        let dt = Duration::from_secs_f32(egui_ctx.input(|i| i.unstable_dt));
+        let mut offset = egui::vec2(-8.0, 32.0);
+
+        for (i, notification) in notifications
+            .iter_mut()
+            .enumerate()
+            .filter(|(_, n)| n.ttl > Duration::ZERO)
```

Comment:
Oh, ttl is only for the toast part? Then maybe call it `toast_ttl` and add a docstring to it